index.js

```diff
@@ -1687,7 +1742,58 @@
   //. //
   //. //   The value at position 1 is not a member of ‘{ x :: FiniteNumber, y :: FiniteNumber }’.
   //. ```
-  function RecordType(fields) {
+  //.
+  //. ##### Extending record types
+  //.
+  //. By using the `parent` argument of the RecordType constructor, one can
+  //. define record types which must contain all the same properties of
+  //. another. For example, one could define `Poinst3D` as a refinement of
+  //. `Point` from the previous example:
+  //.
+  //. ```js
+  //. //    Point3D :: Type
+  //. const Point3D = $.RecordType(Point, {z: $.FiniteNumber});
+  //.
+  //. //    a :: Point
+  //. const a = {x: 0, y: 0};
+  //.
+  //. //    b :: Point, Point3D
+  //. const b = {x: 3, y: 4, z: 5};
+  //.
+  //. Point.validate(a)   // => Right(a)
+  //. Point3D.validate(a) // => Left({propPath: ['z'], value: undefined})
+  //.
+  //. Point.validate(b)   // => Right(b)
+  //. Point3D.validate(b) // => Right(b)
```

Comment:
A few minor complaints:

  - I'd like to see semicolons here;
  - I'd like to see (at least) two spaces before the `//` beginning a line comment; and
  - I'd like to see `Right({x: 0, y: 0})`/`Right({x: 3, y: 4, z: 5})` rather than `Right(a)`/`Right(b)`.